runtime/java_test.go

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package runtime
+
+import (
+	"testing"
+
+	. "github.com/mikelangelo-project/capstan/testing"
+	. "gopkg.in/check.v1"
+	"gopkg.in/yaml.v2"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type javaSuite struct {
+}
+
+var _ = Suite(&javaSuite{})
+
+func (*javaSuite) TestGetBootCmd(c *C) {
+	// Simulate openjdk8-zulu-compact1's meta/run.yaml being parsed.
+	cmdConfs := map[string]*CmdConfig{
+		"openjdk8-zulu-compact1": &CmdConfig{
+			RuntimeType:      Native,
+			ConfigSetDefault: "java",
+			ConfigSets: map[string]Runtime{
+				"java": nativeRuntime{
+					BootCmd: "/java.so",
+					CommonRuntime: CommonRuntime{
+						Env: map[string]string{
+							"XMS":       "512m",
```

Comment:
i agree with your point on the fact that we are checking here that the boot command is properly built. My main concern initially way that we for example remove the `XMX` and `XMS` options - this test will still be passing, however the actual Java package will no longer work with this.

As this is not actually something that should go into this PR, we can just open an issue to come back to it in the future, should the need arise.